loyalty_mass_mailing/models/loyalty_program.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2021 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+import ast
+
+from odoo import api, fields, models
+from odoo.osv import expression
+
+
+class LoyaltyProgram(models.Model):
+    _inherit = "loyalty.program"
+
+    mailing_ids = fields.One2many(
+        comodel_name="mailing.mailing",
+        inverse_name="program_id",
+        string="Mailings",
+        copy=False,
+    )
+    mailing_count = fields.Integer(
+        compute="_compute_mailing_count", string="Mailing count"
+    )
+
+    @api.depends("mailing_ids")
+    def _compute_mailing_count(self):
+        mailing_data = self.env["mailing.mailing"].read_group(
+            [("program_id", "in", self.ids)], ["program_id"], ["program_id"]
+        )
+        mapped_data = {m["program_id"][0]: m["program_id_count"] for m in mailing_data}
+        for program in self:
+            program.mailing_count = mapped_data.get(program.id, 0)
+
+    def action_mailing_count(self):
+        self.ensure_one()
+        xmlid = "mass_mailing.mailing_mailing_action_mail"
+        model = self.env["ir.model"].search([("model", "=", "res.partner")])
+        partner_domains = [
+            ast.literal_eval(domain)
+            for domain in self.rule_ids.mapped("rule_partners_domain")
+            if domain
+        ]
```

Comment:
You mustn't literally eval this. It breaks as sum as you have a variable in the right part. This is still not perfect if the domain is not normalized:

```suggestion
        domain_list = self.rule_ids.mapped("rule_partners_domain")
        partner_domains = "[" + "'|'," * (len(domain_list) - 1) + ",".join(x[1:-1] for x in domain_list) + "]"
```

Although you can create a computed field at coupon level getting this `partner_applicability_domain` for better encapsulating.

Replace `|` by `&` if all the conditions should be met.

Reply:
I have opted for the computed field, as it seems clearer to me to obtain the necessary data. Thank you!

Reply:
> It breaks as sum as you have a variable in the right part

This will be user generated domains. Odoo itself discards the possibility of variable domains in core: https://github.com/odoo/odoo/blob/4ecb41c5f0fd15017ed837e9965ee8c84c77edee/addons/loyalty/models/loyalty_rule.py#L127

Reply:
OK, if that's the case, then let's go with `literal_eval`, but after that, they have to be added with `expression.OR`

Reply:
Yes, I proposed that before seeing your comment :sweat_smile: